src/overlays/actors/ovl_En_Vm/z_en_vm.c

```diff
@@ -96,62 +141,418 @@ static DamageTable D_808CD564 = {
+            sp22 = BINANG_SUB(this->actor.yawTowardsPlayer - this->unk_218, this->actor.shape.rot.y);
+
+            temp_v0 = Math_Vec3f_Pitch(&this->actor.focus.pos, &player->actor.world.pos);
+            if (temp_v0 > 0x1B91) {
+                temp_v0 = 0x1B91;
+            }
+
+            if ((ABS_ALT(sp22) <= 0x2710) && (temp_v0 >= 0xE38) && (Player_GetMask(globalCtx) != PLAYER_MASK_STONE)) {
+                func_808CC5C4(this);
+            }
+        }
+    }
+}
+
+void func_808CC5C4(EnVm* this) {
+    Animation_PlayLoopSetSpeed(&this->skelAnime, &object_vm_Anim_000068, 2.0f);
+    this->unk_214 = 10;
+    this->actionFunc = func_808CC610;
+}
+
+void func_808CC610(EnVm* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    s16 sp3A;
+    s16 sp38;
+
+    SkelAnime_Update(&this->skelAnime);
+
+    sp38 = Math_Vec3f_Pitch(&this->actor.focus.pos, &player->actor.world.pos);
+    sp38 = CLAMP_MAX(sp38, 0x1B91);
+
+    sp3A = BINANG_ADD((s32)((this->unk_21C - this->actor.xzDistToPlayer) * 60.0f), 4000);
+    sp3A = CLAMP_MAX(sp3A, 0x1000);
+
+    Math_SmoothStepToS(&this->unk_216, sp38, 10, 4000, 0);
+
+    if ((sp38 < 0xAAA) || (sp3A <= 0)) {
+        func_808CC420(this);
+    } else if (Math_ScaledStepToS(&this->unk_218, BINANG_SUB(this->actor.yawTowardsPlayer, this->actor.shape.rot.y),
+                                  sp3A)) {
+        this->unk_214--;
+        if (this->unk_214 == 0) {
+            EffectSsDeadDd_Spawn(globalCtx, &this->unk_228, &gZeroVec3f, &gZeroVec3f, &D_808CD5BC, &D_808CD5C0, 150,
+                                 -25, 16, 20);
+            func_808CC788(this);
+        }
+    }
+}
+
+void func_808CC788(EnVm* this) {
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_BIMOS_AIM);
+    Animation_Change(&this->skelAnime, &object_vm_Anim_000068, 3.0f, 3.0f, 7.0f, 2, 0.0f);
+    this->unk_214 = 305;
+    this->unk_220 = 0.06f;
+    this->colliderTris.base.atFlags &= ~AT_HIT;
+    this->actionFunc = func_808CC820;
+}
+
+void func_808CC820(EnVm* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    s16 sp32;
+    f32 temp_f2;
+
+    if (SkelAnime_Update(&this->skelAnime)) {
+        this->skelAnime.curFrame = this->skelAnime.startFrame;
+    }
+
+    sp32 = Math_Vec3f_Pitch(&this->actor.focus.pos, &player->actor.world.pos);
+    sp32 = CLAMP_MAX(sp32, 0x1B91);
+
+    if (this->colliderTris.base.atFlags & AT_HIT) {
+        this->colliderTris.base.atFlags &= ~AT_HIT;
+        this->unk_214 = 0;
+        if (this->unk_220 > 0.01f) {
+            this->unk_220 = 0.01f;
+        }
+    }
+
+    if ((this->unk_216 < 0xAAA) || (Player_GetMask(globalCtx) == PLAYER_MASK_STONE) || (this->unk_214 == 0)) {
+        this->unk_214 = 0;
+        if (Math_StepToF(&this->unk_220, 0.0f, 0.003f)) {
+            this->unk_210 = 0;
+            this->unk_224 = 0.0f;
+            func_808CC420(this);
+        }
+    } else {
+        this->unk_214--;
+        if (this->unk_214 < 301) {
```

Comment:
```suggestion
        if (this->unk_214 <= 300) {
```
probably will make more sense.